app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java

```diff
@@ -250,80 +251,113 @@ private void onUpdatedShouldRunDecision(boolean newShouldRunDecision) {
+
+         @Override
+         protected Void doInBackground(Void... voids) {
+             SyncthingService syncthingService = refSyncthingService.get();
+             if (syncthingService == null) {
+                 cancel(true);
+             }
+             try {
+                 syncthingService.mConfig = new ConfigXml(syncthingService);
+                 syncthingService.mConfig.updateIfNeeded();
+             } catch (ConfigXml.OpenConfigException e) {
+                 syncthingService.mNotificationHandler.showCrashedNotification(R.string.config_create_failed, true);
+                 synchronized (syncthingService.mStateLock) {
+                     syncthingService.onServiceStateChange(State.ERROR);
+                 }
+                 cancel(true);
+             }
+             return null;
+         }
+
+         @Override
+         protected void onPostExecute(Void aVoid) {
+             // Get a reference to the service if it is still there.
+             SyncthingService syncthingService = refSyncthingService.get();
+             if (syncthingService != null) {
+                 syncthingService.onStartupTaskCompleteListener();
+             }
+             return;
+         }
+     }
+
+     /**
+      * Callback on {@link StartupTask#onPostExecute}.
+      */
+     private void onStartupTaskCompleteListener() {
+         if (mApi == null) {
+             mApi = new RestApi(this, mConfig.getWebGuiUrl(), mConfig.getApiKey(),
+                                 this::onApiAvailable, () -> onServiceStateChange(mCurrentState));
+             Log.i(TAG, "Web GUI will be available at " + mConfig.getWebGuiUrl());
+         }
+
+         // Start the syncthing binary.
+         if (mSyncthingRunnable != null || mSyncthingRunnableThread != null) {
+             Log.e(TAG, "onStartupTaskCompleteListener: Syncthing binary lifecycle violated");
+             return;
+         }
+         mSyncthingRunnable = new SyncthingRunnable(this, SyncthingRunnable.Command.main);
+         mSyncthingRunnableThread = new Thread(mSyncthingRunnable);
+         mSyncthingRunnableThread.start();
+
+         /**
+          * Wait for the web-gui of the native syncthing binary to come online.
+          *
+          * In case the binary is to be stopped, also be aware that another thread could request
+          * to stop the binary in the time while waiting for the GUI to become active. See the comment
+          * for {@link SyncthingService#onDestroy} for details.
+          */
+         if (mPollWebGuiAvailableTask == null) {
+             mPollWebGuiAvailableTask = new PollWebGuiAvailableTask(
+                     this,
+                     getWebGuiUrl(),
+                     mConfig.getApiKey(),
+                     result -> {
+                 Log.i(TAG, "Web GUI has come online at " + mConfig.getWebGuiUrl());
+                 if (mApi != null) {
+                     mApi.readConfigFromRestApi();
+                 }
+             });
```

Comment:
The indentation here is strange, either do:
```
new Foo(bar, baz, nuff, squeek -> {
   blah
});
```
or

```
new Foo(
   bar, baz, nuf, squeek -> {
       faf();
   }
);
```

Reply:
Ok, agreed. Second suggestion looks better than my original indent :)